sentry_sdk/attachments.py

```diff
@@ -0,0 +1,51 @@
+import os
+import mimetypes
+
+from sentry_sdk._types import MYPY
+from sentry_sdk.envelope import Item
+
+if MYPY:
+    from typing import Optional
+
+
+class Attachment(object):
+    def __init__(
+        self,
+        bytes=None,  # type: Optional[bytes]
+        filename=None,  # type: Optional[str]
+        path=None,  # type: Optional[str]
+        content_type=None,  # type: Optional[str]
+        add_to_transactions=False,  # type: bool
+    ):
+        # type: (...) -> None
+        if bytes is None and path is None:
+            raise TypeError("path or raw bytes required for attachment")
+        if filename is None and path is not None:
+            filename = os.path.basename(path)
+        if filename is None:
+            raise TypeError("filename is required for attachment")
+        if content_type is None:
+            content_type = mimetypes.guess_type(filename)[0]
+        self.bytes = bytes
+        self.filename = filename
+        self.path = path
+        self.content_type = content_type
+        self.add_to_transactions = add_to_transactions
+
+    def to_envelope_item(self):
+        # type: () -> Item
+        """Returns an envelope item for this attachment."""
+        payload = self.bytes
+        if payload is None:
+            with open(self.path, "rb") as f:
+                payload = f.read()
```

Comment:
Is bubbling up the errors from these safe and desirable (I'm just not familiar enough with the sdk to know)

Reply:
Good question, probably we want to catch it and log. Going to change.

Reply:
Changed.